### PR TITLE
How to write tests for gluster-block using Glusto framework

### PR DESCRIPTION
**Pull request by bhumikagoyal**
This doc explains how to write gluster-block test cases using the glusto framework. Some examples included for better understanding.

### CODE REVIEW

File: docs/glusto_framework.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+The aim of this document is to help in getting started with automating test cases for gluster-block. Therefore, this includes very basic test cases.
+
+## Installing
+
+To automate test cases using glusto, we need to install glusto and other libraries/modules specific to glusterfs and gluster-block.
+Follow this link for the setup: https://github.com/gluster/glusto-tests
+
+## Running the tests
+
+Before writing gluster-block test cases, create 3-4 virtual machines which will serve as clients and servers.
```
Comment (sankarshanmukhopadhyay) [inline docs/glusto_framework.md]: Suggestion; adding environment detail for the machine instances required.

File: docs/glusto_framework.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+We can start writing the test cases now.
+
+The test libraries for block are under glustolibs-gluster package which we have already installed. Import these and other required libraries for writing test cases.
+Here are a few examples of how test cases can be automated for gluster-block.
+
+### Few Examples
+
+@runs_on decorator is used to define the volume types for test cases. Multiple volume types can be given at the same time.
+
+In the below example, the setUpClass method creates a glusterfs volume of type repicated using the information from the config file.
```
Comment (sankarshanmukhopadhyay) [inline docs/glusto_framework.md]: 'replicated'

File: docs/glusto_framework.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+For this example I am assuming that glusterd and gluster-blockd are already started on the servers but they can be started/restarted in the test cases too as you can see below.
```
Comment (sankarshanmukhopadhyay) [inline docs/glusto_framework.md]: For this example, we assume that the glusterd and gluster-blockd srevices are already running on the servers. However, they can be started/re-started in the tests cases as are shown in examples below.

**Review by sankarshanmukhopadhyay — COMMENTED** [review]
Suggest changing the PR title to read "How to write tests for gluster-block using Glusto framework" or, similar.

**Comment by pkalever** [comment]
This didn't make into the project, closing this now.

**Pull request closed without merging**